src/ot_orchestration/utils/batch.py

```diff
@@ -26,99 +36,113 @@ def create_container_runnable(
 
     Returns:
         batch_v1.Runnable: The container runnable.
+
+
     """
-    container = batch_v1.Runnable.Container(
-        image_uri=image, entrypoint="/bin/sh", commands=commands, **kwargs
+    runnable = Runnable()
+    runnable.container = Runnable.Container(
+        image_uri=image, commands=commands, **kwargs
     )
-    return batch_v1.Runnable(container=container)
+    return runnable
 
 
 def create_task_spec(
-    image: str, commands: list[str], **kwargs: Any
-) -> batch_v1.TaskSpec:
+    image: str,
+    commands: list[str],
+    resource_specs: Batch_Resource_Specs,
+    task_specs: Batch_Task_Specs,
+    **kwargs: Any,
+) -> TaskSpec:
     """Create a task for a Batch job.
 
     Args:
         image (str): The Docker image to use.
         commands (list[str]): The commands to run in the container.
+        resource_specs (Batch_Resource_Specs): The specification of the resources for the task.
+        task_specs (Batch_Task_Specs): The specification of the task.
         **kwargs (Any): Any additional parameter to pass to the container runnable
 
     Returns:
         batch_v1.TaskSpec: The task specification.
     """
-    task = batch_v1.TaskSpec()
-    task.runnables = [create_container_runnable(image, commands, **kwargs)]
+    task = TaskSpec()
+    task.runnables = [create_container_runnable(image, commands=commands, **kwargs)]
+    resources = ComputeResource()
+    resources.cpu_milli = resource_specs["cpu_milli"]
+    resources.memory_mib = resource_specs["memory_mib"]
+    resources.boot_disk_mib = resource_specs["boot_disk_mib"]
+    task.compute_resource = resources
+
+    task.max_retry_count = task_specs["max_retry_count"]
+    task.max_run_duration = Duration(
+        seconds=time_to_seconds(task_specs["max_run_duration"])
+    )
```

Comment:
```suggestion
    task.max_run_duration = f"{time_to_seconds(task_specs['max_run_duration'])}s"
```

I think you can get away with a string in here (or even return the string with the `s` straight away from `time_to_seconds`). Not sure how good it is to import the underlying protobuffer types from google cloud. This is also simpler.

But I did not check if it works, it might complain.

Reply:
you are probably right - 
```
       max_run_duration (google.protobuf.duration_pb2.Duration):
            Maximum duration the task should run before being
            automatically retried (if enabled) or automatically failed.
            Format the value of this field as a time limit in seconds
            followed by ``s``—for example, ``3600s`` for 1 hour. The
            field accepts any value between 0 and the maximum listed for
            the ``Duration`` field type at
            https://protobuf.dev/reference/protobuf/google.protobuf/#duration;
            however, the actual maximum run time for a job will be
            limited to the maximum run time for a job listed at
            https://cloud.google.com/batch/quotas#max-job-duration.
```
but this is so not obvious and offends type definition in that field, which is sad, as I would expect that the `protobuf.MESSAGE` will map to a string type.